cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1860,8 +1868,16 @@ func (p *BicepProvider) modulePath() string {
 func (p *BicepProvider) ensureParameters(
 	ctx context.Context,
 	template azure.ArmTemplate,
-	parameters azure.ArmParameters,
 ) (azure.ArmParameters, error) {
+	if p.ensureParamsInMemoryCache != nil {
+		return p.ensureParamsInMemoryCache, nil
```

Comment:
Would you want to return a clone here as well so that if the caller modifies the map it doesn't hork the cached copy?  Just wondering - given you do an explicit clone when you set it.